examples/spark2/src/main/scala/org/apache/carbondata/benchmark/CDCBenchmark.scala

```diff
@@ -0,0 +1,256 @@
+      .map {x => CDC(s"newOrder${System.currentTimeMillis()}", s"customer$x", updateDate, 1)
+      }.toDS().toDF()
+    ds1.union(ds2)
+  }
+
+  def main(args: Array[String]): Unit = {
+    import org.apache.spark.sql.CarbonSession._
+    val rootPath = new File(this.getClass.getResource("/").getPath
+                            + "../../../..").getCanonicalPath
+    val storeLocation = s"$rootPath/examples/spark2/target/store"
+    val master = Option(System.getProperty("spark.master"))
+      .orElse(sys.env.get("MASTER"))
+      .orElse(Option("local[8]"))
+
+    val spark = SparkSession
+      .builder()
+      .master(master.get)
+      .enableHiveSupport()
+      .config("spark.driver.host", "127.0.0.1")
+      .getOrCreateCarbonSession(storeLocation)
+    spark.sparkContext.setLogLevel("warn")
+
+    spark.sql("drop table if exists dw_order")
+    spark.sql("drop table if exists ods_order")
+
+    // prepare base data for first day
+    val df = generateDataForDay0(
+      sparkSession = spark,
+      numOrders = numOrders,
+      startDate = Date.valueOf("2018-05-01"))
+
+    spark.sql(s"drop table if exists dw_order")
+    df.write
+      .format("carbondata")
+      .option("tableName", "dw_order")
+      .mode(SaveMode.Overwrite)
+      .save()
+
+    var startDate = Date.valueOf("2018-05-01")
+    var state = 2
+    var updateTime = 0L
+
+    if (printDetail) {
+      println("## day0")
+      spark.sql("select * from dw_order").show(100, false)
+    }
+
+    for (i <- 1 to numDays) {
+      // prepare for incremental update data for day-i
+      val newDate = new Date(DateUtils.addDays(startDate, 1).getTime)
+      val cdc = generateDailyCDC(
+        sparkSession = spark,
+        numUpdatedOrders = numUpdateOrdersDaily,
+        startDate = startDate,
+        updateDate = newDate,
+        newState = state,
+        numNewOrders = newNewOrdersDaily)
+      cdc.write
+        .format("carbondata")
+        .option("tableName", "ods_order")
+        .mode(SaveMode.Overwrite)
+        .save()
+
+      if (printDetail) {
+        println(s"day$i CDC")
+        spark.sql("select * from ods_order").show(100, false)
+      }
+
+      // update dw table using CDC data
+      val start = System.nanoTime()
+      hive_solution(spark)
+      // carbon_solution(spark)
+      val end = System.nanoTime()
+      updateTime += end - start
+
+      if (printDetail) {
+        println(s"day$i result")
+        spark.sql("select * from dw_order").show(100, false)
+      }
+
+      startDate = newDate
+      state = state + 1
+    }
+
+    println(s"simulated $numDays days, total process time takes ${updateTime / 1000 / 1000} ms")
+    spark.close()
+  }
+
+  /**
+   * Typical solution when using hive
+   * This solution uses INSERT OVERWRITE to rewrite the whole table every day
+   */
+  private def hive_solution(spark: SparkSession) = {
+    spark.sql(
+      """
+        | insert overwrite table dw_order
```

Comment:
Maybe it is better to create a partition table on date column and insert overwrite only on the partition would be better.

Reply:
I will do it in another PR.